src/reaction_lammps_mupt/reaction_template_builder/reaction_template_pipeline/util.py

```
@@ -0,0 +1,167 @@
+            return data_smiles_list, False  # or raise a ValueError
+        # Generate canonical SMILES and compare them
+        canonical_smi_1 = Chem.MolToSmiles(mol1, canonical=True)
+        canonical_smi_2 = Chem.MolToSmiles(mol2, canonical=True)
+
+        if canonical_smi_1 == canonical_smi_2:
+            return data_smiles_list, True
+    
+    data_smiles_list.append(mol_smi_2)
+    return data_smiles_list, False
+
+def format_detected_reactions_dict(detected_reactions, non_monomer_molecules_to_retain = None):
```

Ruff (0.14.14)

[warning] Unused function argument: `non_monomer_molecules_to_retain`

(ARG001)